src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Patterns.cs

```diff
@@ -133,12 +133,25 @@ private bool IsIrrefutablePatternTest(BoundExpression test)
 
         private BoundExpression CompareWithConstant(BoundExpression input, BoundExpression boundConstant)
         {
-            return _factory.StaticCall(
-                _factory.SpecialType(SpecialType.System_Object),
-                "Equals",
-                _factory.Convert(_factory.SpecialType(SpecialType.System_Object), boundConstant),
-                _factory.Convert(_factory.SpecialType(SpecialType.System_Object), input)
-                );
+            var systemObject = _factory.SpecialType(SpecialType.System_Object);
+            if (boundConstant is BoundLiteral bl && bl.ConstantValue.IsNull)
```

Comment:
Looking again, perhaps because the pattern is lowered that the constant *is* always a literal. If that is the case, you shouldn't have to test for it.